### PR TITLE
Change summary to cve_description

### PR DESCRIPTION
**Pull request by JakubFrejlach**
# [OSIDB-2785] [Change summary to cve_description]

## Checklist:

- [x] Linting passed
- [x] Type checks passed
- [x] Tests suite passed
- [x] Commits consolidated
- [x] Changelog updated
- [ ] ~~Test cases added/updated~~
- [x] Jira ticket updated

## Summary:

We are already showing `Flaw.summary` as `Description` field. This PR is a final step to ditch `Flaw.summary` usage completely and start using `Flaw.cve_description` which will be introduced in the OSIDB 4.0.0. With `summary` to `cve_description` change, `requires_summary` is also changing to `requires_cve_description`

## Changes:
 
* complete transition to `Flaw.cve_description` and `Flaw.requires_cve_description` usage

## Considerations:

This PR needs to be merged only after OSIDB-2740 is addressed!

### CODE REVIEW

**Comment by JakubFrejlach** [comment]
**NOTE**: Feel free to review, but this PR **must not be** merged before OSIDB-2740 is addressed. Also the tests are currently failing because the client needs to be regenerated from the new OpenAPI schema, this will be also done once OSIDB-2740 is addressed.

**Review by C-Valen — COMMENTED** [review]
Couple of minor considerations

File: src/components/FlawForm.vue [hunk 1]
```diff
@@ -106,7 +106,7 @@ const onSubmit = async () => {
   }
 };
 
-const showSummary = ref(flaw.value.summary && flaw.value.summary.trim() !== '');
+const showDescription = ref(flaw.value.cve_description && flaw.value.cve_description.trim() !== '');
```
[the remark this answers is not in the document]
Comment (JakubFrejlach) [reply src/components/FlawForm.vue:110]: The thing is, the actual field name should be `description`, OSIDB is currently just using `cve_description` so it won't be confused with the other OSIDB `description` field which changed to `comment_zero`, ideally in some next OSIDB version, we would like to stick to the `description` field name.

File: CHANGELOG.md [hunk 2]
```diff
@@ -8,6 +8,9 @@
 * Add CVE Require Description for AdvancedSearch (`OSIDB-2624`)
 * Support for references and acknowledgements on flaw creation (`OSIDB-2319`)
 
+### Changed
+* Use cve_description/requires_cve_description instead of summary/requires_summary from OSIDB (`OSIDB-2785`)
```
[the remark this answers is not in the document]
Comment (JakubFrejlach) [reply CHANGELOG.md:21]: Hmm, actually I think this changelog shouldn't be there at all, it is just an internal change.

File: CHANGELOG.md [hunk 3]
```diff
@@ -14,9 +14,6 @@
 * Advanced search on emptiness for CVE ID (`OSIDB-2806`)
 * Embedded Alerts on top of the Flaw form (`OSIDB-1211`)
 
-### Changed
-* Use comment_zero instead of description from OSIDB (`OSIDB-2784`)
```
Comment (MrMarble) [inline CHANGELOG.md:18]: Why are we deleting the chagelog from OSIDB-2784 and not adding a new one for OSIDB-2785?

**Review by MrMarble — APPROVED** [review]
Other than the changelog think it looks good

File: CHANGELOG.md (hunk 3 above)
Comment (JakubFrejlach) [reply CHANGELOG.md:18]: It was a reaction on the Carlos previous point here https://github.com/RedHatProductSecurity/osim/pull/242#discussion_r1625552686 

This is not a user facing thing, for user nothing changes, as it is already a `Description` field, we just change the field which is the source for the data and the same case was for the OSIDB-2784, user already sees `Comment #0` in OSIM, we just changed the field source, so there should have been no changelog entries, so just retrospectively deleting this one.

**Pull request merged**